core/src/lib.rs

```diff
@@ -112,24 +108,71 @@ pub type DefaultFileService = FileServiceImpl<
     DefaultFileEncryptionService,
 >;
 
+macro_rules! impl_get_variants {
```

Comment:
Maybe worth dropping some comments here explaining what's going on. Also let's move this to the bottom of this file.

Reply:
sure thing